pkg/resources/schema.go

```diff
@@ -133,10 +131,15 @@ func ReadSchema(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if _, ok := d.GetOk("data_retention_days"); ok || (!ok && retentionTime != int64(database.RetentionTime)) {
```

Comment:
`!ok` part is not needed because we get to the second operand only if ok is not true

Reply:
I am not convinced why we are setting the data retention for `!ok && retentionTime != int64(database.RetentionTime)` case. It is inconsistent:
1. for the simple case where we just do not fill the config for schema and want to use the database one, the data_retention_days on schema won't be filled.
2. for the case when someone alters the schema by hand, we will set this.

Reply:
My thinking was that: if `retentionTime == database.RetentionTime`, it means that it's the same as default so the value shouldn't be set, but when they're not the same it means there was some manual change done on the Snowflake side, and terraform should correct it to the value from the config.

Reply:
Yes, this is good, I was not precise enough, I only meant setting it in state (added a comment in test).